rfcs/0003-build-contents-diff.md

```diff
@@ -0,0 +1,160 @@
+### Build Contents Diff RFC 
+
+## Context
+In order to make decisions about what to do with any given build generated by TBS, users need to compare a new build’s contents with the contents of other builds.
+
+## Goals
+Create a new command, `kp build diff`, for printing the difference between two builds. The command will take one required argument and two optional arguments.
+ 
+Examples:
+`kp build diff image-name`
+Compares the two most recent builds
+`kp build diff image-name build-number1`
+Compares the stated build and the build minus 1
+`kp build diff image-name build-number1 build-number2`
+Compares the two specified builds
+ 
+Three use cases that this feature should ideally satisfy:
+ 
+- As a DevOps engineer, when I see a new build has been created in a lower environment, I want to evaluate what benefit it provides to my team for me to promote this build to an upper environment. 
+ 
+- As a DevOps engineer or App Developer, when I am told there is a problem with a specific build (new build failing in CI or old build failing in production), I want to know what’s different about the problem build than the last one that worked. 
+
+- As an App Developer, I want to see the difference between the commit I built locally with pack yesterday and the build that kpack built for me today based on my commit. 
+ 
+Implications: 
+- User will need to compare their most recent build to the older build in production
+i.e. builds being compared are non-adjacent 
+- Because the builds are non-adjacent, many things are likely to have changed in the contents and there will have been many “Reasons” that caused the intervening builds. Therefore, it doesn’t make a lot of sense to print the “Reason” as part of the diff
+- I assume that it is easy / possible for users to access the build number associated with a container in production (perhaps there is another build identifier we could use?)
```

Comment:
Given that the use case this feature seems to address is comparing a given build against one in production, seems like we should support providing an image by its tag or digest to compare against a particular build.  I know we talked about this a couple of weeks ago, but may be worth discussing further in a working group setting.

Reply:
Given that it is probably likely that the image to be diffed against may no longer be available or easily retrievable as a build number it seems important to allow the diffing to be performed against a specific image via tag or digest.

Reply:
This is especially true if we want to enable the pack usecase listed above.

Reply:
Great call out, thank you